magda-typescript-common/src/Registry.ts

```diff
@@ -48,10 +50,31 @@ export default class Registry {
             .catch(createServiceError);
     }
 
+    async forEachRecord(callback: (record: Record) => Promise<any>, aspect?: Array<string>, optionalAspect?: Array<string>, dereference?: boolean) {
+        console.log(aspect);
+        const operation = (pageToken: string) => () => this.recordsApi.getAll(aspect, optionalAspect, pageToken, undefined, undefined, dereference);
+        let records;
+        let pageToken = undefined;
+        do {
+            records = await retry(operation(pageToken), this.secondsBetweenRetries, this.maxRetries, (e, retriesLeft) => console.log(formatServiceError('Failed to GET records.', e, retriesLeft)))
+                .then(result => result.body)
+                .catch(createServiceError);
+            await Promise.all((<any>records).records.map(callback));
```

Comment:
Why do the typescript definitions expect the type `records: Array<RecordSummary>`?

AFAICT this is because of a problem with the generated typescript here: https://github.com/TerriaJS/magda/blob/60af342/magda-typescript-common/src/generated/registry/api.ts#L1124